unit/java_bytecode/java_virtual_functions/virtual_functions.cpp

```diff
@@ -115,6 +115,55 @@ SCENARIO(
       }
 
       REQUIRE(found_function);
+
+      WHEN("basic blocks are instrumented")
+      {
+        optionst options;
+        options.set_option("cover", "location");
+        REQUIRE_FALSE(
+          instrument_cover_goals(
+            options, new_table, new_goto_functions, null_output));
+
+        auto function = new_goto_functions.function_map.find(function_name);
+        REQUIRE(function != new_goto_functions.function_map.end());
+
+        const goto_programt &goto_program = function->second.body;
+        auto it = goto_program.instructions.begin();
+        REQUIRE(it != goto_program.instructions.end());
+
+        // Skip the first instruction which is a declaration with no location
+        ++it;
```

Comment:
My feeling is you should do `std::next(goto_program.instructions.begin())` but then the require would be that the program includes at least two source locations, but currently you check the first one is not the end, then advance one and then dereference it, which might be a problem as it might be at the end. Using `std::next` when init'ing would prevent this.